src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -84,84 +55,138 @@ OTARequestorInterface * GetRequestorInstance()
     return globalOTARequestorInstance;
 }
 
-void StartDelayTimerHandler(chip::System::Layer * systemLayer, void * appState)
+static void LogQueryImageResponse(const QueryImageResponse::DecodableType & response)
 {
-    VerifyOrReturn(appState != nullptr);
-    static_cast<OTARequestor *>(appState)->ConnectToProvider();
-}
-
-void OnQueryImageFailure(void * context, EmberAfStatus status)
-{
-    ChipLogDetail(SoftwareUpdate, "QueryImage failure response %" PRIu8, status);
+    ChipLogDetail(SoftwareUpdate, "QueryImageResponse:");
+    ChipLogDetail(SoftwareUpdate, "  status: %" PRIu8 "", response.status);
+    if (response.delayedActionTime.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  delayedActionTime: %" PRIu32 " seconds", response.delayedActionTime.Value());
+    }
+    if (response.imageURI.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  imageURI: %.*s", static_cast<int>(response.imageURI.Value().size()),
+                      response.imageURI.Value().data());
+    }
+    if (response.softwareVersion.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  softwareVersion: %" PRIu32 "", response.softwareVersion.Value());
+    }
+    if (response.softwareVersionString.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  softwareVersionString: %.*s",
+                      static_cast<int>(response.softwareVersionString.Value().size()),
+                      response.softwareVersionString.Value().data());
+    }
+    if (response.updateToken.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  updateToken: %zu", response.updateToken.Value().size());
+    }
+    if (response.userConsentNeeded.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  userConsentNeeded: %d", response.userConsentNeeded.Value());
+    }
+    if (response.metadataForRequestor.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  metadataForRequestor: %zu", response.metadataForRequestor.Value().size());
+    }
 }
 
-void OnConnectionFailure(void * context, NodeId deviceId, CHIP_ERROR error)
+// TODO: Add unit tests for parsing BDX URI
+CHIP_ERROR OTARequestor::ParseBdxUri(CharSpan uri, NodeId & nodeId, CharSpan & fileDesignator)
```

Comment:
Why not add the unit tests today?

Reply:
Given that we're trying to ask platform vendors to develop, this is one of the last changes that affects what they need to pass in to their application (the IP address). I wanted to get this in first and then address unit testing later.